static/app/views/projectInstall/issueAlertNotificationOptions.tsx

```diff
@@ -0,0 +1,277 @@
+  const [integration, setIntegration] = useState<OrganizationIntegration | undefined>(
+    undefined
+  );
+  const [channel, setChannel] = useState<string | undefined>(undefined);
+
+  const api = useApi();
+  const organization = useOrganization();
+
+  type Props = {
+    actionMatch: string | undefined;
+    conditions: {id: string; interval: string; value: string}[] | undefined;
+    frequency: number | undefined;
+    name: string | undefined;
+    projectSlug: string;
+    shouldCreateRule: boolean | undefined;
+  };
+
+  const createNotificationAction = useCallback(
+    ({
+      shouldCreateRule,
+      projectSlug,
+      name,
+      conditions,
+      actionMatch,
+      frequency,
+    }: Props) => {
+      const isCreatingIntegrationNotification = actions.find(
+        action => action === MultipleCheckboxOptions.INTEGRATION
+      );
+      if (
+        !organization.features.includes(
+          'messaging-integration-onboarding-project-creation'
+        ) ||
+        !shouldCreateRule ||
+        !isCreatingIntegrationNotification
+      ) {
+        return undefined;
+      }
+
+      let integrationAction;
+      switch (provider) {
+        case 'slack':
+          integrationAction = [
+            {
+              id: IssueAlertActionType.SLACK,
+              workspace: integration?.id,
+              channel: channel,
+            },
+          ];
+          break;
+        case 'discord':
+          integrationAction = [
+            {
+              id: IssueAlertActionType.DISCORD,
+              server: integration?.id,
+              channel_id: channel,
+            },
+          ];
+          break;
+        case 'msteams':
+          integrationAction = [
+            {
+              id: IssueAlertActionType.MS_TEAMS,
+              team: integration?.id,
+              channel: channel,
+            },
+          ];
+          break;
+        default:
+          return undefined;
+      }
+
+      return api.requestPromise(`/projects/${organization.slug}/${projectSlug}/rules/`, {
+        method: 'POST',
+        data: {
+          name,
+          conditions,
+          actions: integrationAction,
+          actionMatch,
+          frequency,
+        },
+      });
+    },
+    [
+      actions,
+      api,
+      provider,
+      integration,
+      channel,
+      organization.features,
+      organization.slug,
+    ]
+  );
+
+  return {
+    createNotificationAction,
+    notificationProps: {
+      actions,
+      provider,
+      integration,
+      channel,
+      setActions,
+      setProvider,
+      setIntegration,
+      setChannel,
+    },
+  };
+}
+
+export default function IssueAlertNotificationOptions(
+  notificationProps: IssueAlertNotificationProps
+) {
+  const organization = useOrganization();
+  const {actions, provider, setActions, setIntegration, setProvider} = notificationProps;
+
+  const messagingIntegrationsQuery = useApiQuery<OrganizationIntegration[]>(
```

Comment:
Any reason this stuff shouldn't be part of the `useCreateNotificationAction` hook?

Reply:
I think it makes sense to leave this here because it's purpose is determining what to render (i.e. the setup messaging integration button vs. the notification configuration), while the `useCreateNotificationAction` hook is for setting up the notification configuration (the integration provider, channel, etc.).

The results of `messagingIntegrationsQuery` are only really used in this component, so I think it makes it unnecessarily complicated for the hook to return the results to `createProject` just for `createProject` to pass it down to its child components.

Reply:
Hm actually I take this back, since `messagingIntegrationsQuery.data` is used to create the `providersToIntegrations` mapping and determine the initial values of `provider` and `integration`

Reply:
Yeah I think it would be OK for this to be part of the hook.